doc/developers/contributing.rst

```diff
@@ -953,10 +953,14 @@ All logic behind estimator parameters,
 like translating string arguments into functions, should be done in ``fit``.
 
 Parameters can be passed to the ``fit`` method. Generally, these parameters
-should be restricted to variables that have the shape ``n_samples``.
-This allows for all relevant parameters to be sliced cleanly during
-cross-validation. All other parameters should be passed to
-``__init__`` or ``set_params``.
+should be restricted to variables that need to be sliced during
+cross-validation. Consequently, these variables would be either arrays with shape=[N] or
+array-like with shape=[N,D] where N is the number of samples. An example of
+this type of variable would be
+``sample_weight``. This interface allows all data-dependent
+parameters to be sliced cleanly during cross-validation.
+All other parameters should be passed to
+``__init__`` or set after construction using ``set_params``.
```

Comment:
I guess it works fine without consequently. I'll make the change. Good
point on the shape definition also, I'll change it to [N, *] to avoid
confusion. As for the \__init__ vs set_ params, I thought they were
supposed to have the same effect, per the paragraph above explicitly
referencing gridsearch. Could you explain a little more here? Perhaps I'm
not familiar enough with gridsearch.

On Aug 7, 2016 1:45 AM, "Nelson Liu" <notifications@github.com> wrote:

> In doc/developers/contributing.rst
> <https://github.com/scikit-learn/scikit-learn/pull/7156#discussion_r73796196>
> :
>
> > @@ -953,10 +953,14 @@ All logic behind estimator parameters,
> >  like translating string arguments into functions, should be done in ``fit``.
> >
> >  Parameters can be passed to the ``fit`` method. Generally, these parameters
> > -should be restricted to variables that have the shape ``n_samples``.
> > -This allows for all relevant parameters to be sliced cleanly during
> > -cross-validation. All other parameters should be passed to
> > -``__init__`` or ``set_params``.
> > +should be restricted to variables that need to be sliced during
> > +cross-validation. Consequently, these variables would be either arrays with shape=[N] or
> > +array-like with shape=[N,D] where N is the number of samples. An example of
> > +this type of variable would be
> > +``sample_weight``. This interface allows all data-dependent
> > +parameters to be sliced cleanly during cross-validation.
> > +All other parameters should be passed to
> > +``__init__`` or set after construction using ``set_params``.
>
> I think it's worth reminding the users that setting things in __init__ is
> preferred, especially if it's a parameter that effects the way fit
> behaves. I don't think parameters set separately through set_params work
> with GridSearchCV or similar things.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7156/files/c0193304c4304f6cba77749faa86b48964466fda..bf626c9de30b0e51a42d8076748eda54cf8a67dd#r73796196>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKNQ_F4aQNjd73cHiHzj8Aws9UctA1dsks5qdXEEgaJpZM4JeZ4l>
> .
>